warp/src/raygun/mod.rs

```diff
@@ -936,17 +846,17 @@ pub trait RayGun:
         _: Option<String>,
         _: Vec<DID>,
         _: GroupSettings,
-    ) -> Result<Conversation, Error> {
+    ) -> Result<Box<dyn Conversation>, Error> {
         Err(Error::Unimplemented)
     }
 
     /// Get an active conversation
-    async fn get_conversation(&self, _: Uuid) -> Result<Conversation, Error> {
+    async fn get_conversation(&self, _: Uuid) -> Result<Box<dyn Conversation>, Error> {
         Err(Error::Unimplemented)
     }
 
     /// List all active conversations
-    async fn list_conversations(&self) -> Result<Vec<Conversation>, Error> {
+    async fn list_conversations(&self) -> Result<Vec<Box<dyn Conversation>>, Error> {
```

Comment:
same question here as the other PR: can't we just return a `&[Box<..>]` here or `impl Iterator<Item = Box<dyn Conversation>> + '_` here? 🤔

Reply:
Doing that would cause the trait not to be object safe if i recall (same comment on the other PR)